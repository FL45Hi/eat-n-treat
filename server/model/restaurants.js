const mongoose = require("mongoose");

const restaurantSchema = new mongoose.Schema({
  _id: mongoose.Schema.Types.ObjectId,
  restaurantName: {
    type: String,
    required: true,
  },
  restaurantRegistrationStatus: {
    type: Boolean,
  },
  resturanturl: {
    type: String,
  },
  restaurantPhone: {
    type: Number,
  },
  restaurantEmail: {
    type: String,
    lowercase: true,
    trim: true,
    unique: true,
    required: true,
  },

  rCity: {
    type: String,
  },
  rating: {
    type: Number,
  },
  reviews: {
    type: Array,
    of: new mongoose.Schema({
      _id: mongoose.Schema.Types.ObjectId,
      reviewText: String,
      user: { type: mongoose.Schema.Types.ObjectId, ref: "customer" },
      username: String,
    }),
  },
  menus: {
    type: Array,
    of: new mongoose.Schema({
      _id: mongoose.Schema.Types.ObjectId,
      menuName: {
        type: String,
      },
      menuPrice: {
        type: Number,
      },
    }),
  },
});

module.exports = mongoose.model("restaurant", restaurantSchema);
