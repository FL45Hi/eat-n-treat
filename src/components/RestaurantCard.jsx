import React, { useRef, useEffect } from "react";
import { Link } from 'react-router-dom';
import ReactStars from 'react-rating-stars-component';


export default function RestaurantCard(props) {
    var scrollRef = useRef();

    useEffect(() => {
        scrollRef.current.addEventListener('mousewheel', horizontalScroll, false)
    }, [])

    const horizontalScroll = (e) => {
        e = window.event || e;
        var delta = Math.max(-1, Math.min(1, (e.wheelDelta || -e.detail)));
        scrollRef.current.scrollLeft -= (delta * 40); // Multiplied by 40
        e.preventDefault();
    }

    return (
        <div className="card">
            <img
                src="https://mdbootstrap.com/img/new/standard/nature/184.jpg"
                className="card-img-top"
                alt="..."
            />
            <div className="card-body">
                <h5 className="card-title">{props.restaurant.restaurantName}</h5>
                <div className="card-text">
                    <div className="flex-nowrap scrollableDiv" ref={scrollRef}>
                        {
                            props.restaurant.cuisines.map(cuisine => {
                                return (
                                    <div className="chip" style={{ width: "fit-content", display: "inline-block" }}>
                                        { cuisine }
                                    </div>
                                );
                            })
                        }
                    </div> <br />
                    {props.restaurant.location} <br />
                    Ratings: <br />
                    <ReactStars
                        count={5}
                        value={props.restaurant.rating}
                        edit={false}
                        size={24}
                        isHalf={true}
                        emptyIcon={<i className="far fa-star"></i>}
                        halfIcon={<i className="fa fa-star-half-alt"></i>}
                        fullIcon={<i className="fa fa-star"></i>}
                        activeColor="#ffd700"
                    />
                    {props.restaurant.rating}/5
                </div>
                <Link to={`/restaurant/${props.restaurant.restaurantID}`} className="btn btn-primary">VIEW MORE</Link>
            </div>
             
        </div>
    );
}