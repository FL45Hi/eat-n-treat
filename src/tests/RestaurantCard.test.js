import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import restaurants from '../assets/data.json';

import RestaurantCard from '../components/RestaurantCard';

it('Card renders with appropriate restaurant data', () => {
    const tree = renderer.create(<BrowserRouter><RestaurantCard restaurant={ restaurants[0] }/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})