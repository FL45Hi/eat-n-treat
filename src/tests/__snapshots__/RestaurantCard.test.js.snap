// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders with appropriate restaurant data 1`] = `
<div
  className="card"
>
  <img
    alt="..."
    className="card-img-top"
    src="https://mdbootstrap.com/img/new/standard/nature/184.jpg"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Annapoorna
    </h5>
    <div
      className="card-text"
    >
      <div
        className="flex-nowrap scrollableDiv"
      >
        <div
          className="chip"
          style={
            Object {
              "display": "inline-block",
              "width": "fit-content",
            }
          }
        >
          bread-butter
        </div>
        <div
          className="chip"
          style={
            Object {
              "display": "inline-block",
              "width": "fit-content",
            }
          }
        >
          Mexiacan tomatoes
        </div>
        <div
          className="chip"
          style={
            Object {
              "display": "inline-block",
              "width": "fit-content",
            }
          }
        >
          italian burger 
        </div>
      </div>
       
      <br />
      Pune
       
      <br />
      Ratings: 
      <br />
      <div
        className="react-stars-wrapper-"
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          aria-label="add rating by typing an integer from 0 to 5 or pressing arrow keys"
          className=""
          onKeyDown={[Function]}
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
          tabIndex={null}
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": "
    .react-stars-:before {
      position: absolute;
      overflow: hidden;
      display: block;
      z-index: 1;
      top: 0; left: 0;
      width: 50%;
      content: attr(data-forhalf);
      color: #ffd700;
  }",
              }
            }
          />
          <p
            role="status"
            style={
              Object {
                "left": "-200rem",
                "position": "absolute",
              }
            }
          >
            0
          </p>
        </div>
      </div>
      4
      /5
    </div>
    <a
      className="btn btn-primary"
      href="/restaurant/undefined"
      onClick={[Function]}
    >
      VIEW MORE
    </a>
  </div>
</div>
`;
